### PR TITLE
Desktop browser backed by Electron

### PR DESCRIPTION
**Pull request by modulesio**
We originally used Electron for 2d iframes, then Chromium Embedded Framework in-process for performance and reliability.

However, things have changed for Electron and CEF; Electron is well-maintained it looks like painting can now be moved efficiently across the process boundary between Exokit and the electron binary (from my testing).

This PR tries out backing the 2d browser rendering by Electron again instead of CEF, mostly for the case of bug and performance fixes in Exokit Studio.

Fixes https://github.com/exokitxr/studio/issues/13.

### CODE REVIEW

**Comment by modulesio** [comment]
This implementation should be working with studio now.

A few todos before we can ship it:

- [ ] Handle resize
- [ ] Handle rescaling
- [ ] Remove the hefty old desktop CEF builds

**Comment by modulesio** [comment]
I noticed collapsing the console in studio does not collapse the viewport with this branch.

**Comment by modulesio** [comment]
This is interesting for frame sharing: https://github.com/chromium/chromium/blob/fade73660e78869d709556282f1c24c09788a113/gpu/command_buffer/client/shared_image_interface.h

**Comment by modulesio** [comment]
This should be tracking the engine window underneath the html ui now; we just need it appearing on top of it.

File: src/electron-child.js [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+            const s = JSON.stringify(o);
+            let m = new TextEncoder().encode(s);
+            m = new Buffer(m.buffer, m.byteOffset, m.byteLength);
+            const b = Uint32Array.from([TYPES.EVENT, m.length]);
+            const b2 = new Buffer(b.buffer, b.byteOffset, b.byteLength);
+            _parentPortWrite(b2);
+            _parentPortWrite(m);
+          };
+
+          mainWindow = new BrowserWindow({
```
Comment (chrisplatorres) [inline src/electron-child.js]: can specify icon here:

```
mainWindow = new BrowserWindow({
...
 icon: __dirname + ‘/Icon/Icon.icns’
})
```

( related: https://github.com/exokitxr/exokit/issues/995, https://github.com/exokitxr/exokit/issues/526, https://github.com/exokitxr/studio/issues/25 )
reference: https://medium.com/fantageek/changing-electron-app-icon-acf26906c5ad

**Comment by chrisplatorres** [comment]
Found edge case when testing:

studio02

**Comment by modulesio** [comment]
Thanks, is that minimizing or resizing to zero?

**Comment by modulesio** [comment]
I couldn't repro the issue in https://github.com/exokitxr/exokit/pull/1187#issuecomment-502403739, so breaking it out: https://github.com/exokitxr/exokit/issues/1215

**Pull request merged**